tests/sqlparser_mysql.rs

```diff
@@ -176,6 +176,33 @@ fn parse_quote_identifiers() {
     }
 }
 
+#[test]
+fn parse_escaped_string() {
+    let sql = r#"SELECT 'I\'m fine'"#;
+
+    let projection = mysql().verified_only_select(sql).projection;
+    let item = projection.get(0).unwrap();
+
+    match &item {
+        SelectItem::UnnamedExpr(Expr::Value(value)) => {
+            assert_eq!("'I\\'m fine'", value.to_string());
```

Comment:
I would expect this to be

```suggestion
            assert_eq!("'I'm fine'", value.to_string());
```

Reply:
@alamb you're right here.

Looking at the current implementation a `SingleQuotedString` value for `'I''m fine'` will be `I'm fine` as consecutive quote is removed (added back in the formatter). 

I can use the same approach (which seems totally correct) and a `SingleQuotedString` value for `'I\'m fine'` will 'be the same as above `I'm fine`. The only problem here is that this test will be failing
https://github.com/sqlparser-rs/sqlparser-rs/blob/9569d1b21596c150643a212d09607aacd8f60677/src/test_utils.rs#L95

as `SELECT 'I\'m fine'` will never be equals to `SELECT 'I''m fine'`. Also, doesn't seem like we can tweak the formatter as it's unaware of the context

https://github.com/sqlparser-rs/sqlparser-rs/blob/9569d1b21596c150643a212d09607aacd8f60677/src/ast/value.rs#L148

Reply:
Are you saying that the `Display` impl from`to_string()` function is what is adding `\\` back? 

if so, perhaps we could add a check like 

```rust
if let Value::SingleQuotedString(value) = value {
  assert_eq!("I'm fine", value)
} else { 
  unreachable!()
}
```

to show that the value has been properly un-escaped?